### ISSUE TITLE
Feature request: propagate [UnconditionalSuppressMessage] to compiler-generated methods in the nested lexical scope

### ISSUE THREAD

**Issue by agocke** [dotnet/roslyn#60898]

**Comment by agocke** [comment]
cc @vitek-karas @sbomer

**Comment by sbomer** [comment]
I think a relevant question is whether this should happen for any other attributes. For example, it would arguably make sense for `RequiresUnreferencedCode` or other capability attributes to apply to nested scopes too. Maybe not all of these need to be visible to IL analysis tools, but `RequiresUnreferencedCode` at least should be.

**Comment by agocke** [comment]
Yeah, an alternative would be to have some "IL analysis" attribute that allows other attributes to opt-in to the same behavior.

Honestly, there might be some other value here for things like alpha renaming in closure conversion. Right now when the compiler converts nested functions to closures it drops all the attributes from the alpha-renamed type parameters. It would probably make our lives simpler if those were preserved. Since the compiler always does pure alpha renaming of type parameters, there shouldn't be any technical limitations here that I can think of.

**Comment by agocke** [comment]
@jcouv I would be interested in your opinion on accepting this feature set, as I might have time to contribute an implementation for this.

**Comment by jcouv** [comment]
I'd recommend scheduling some time with Jared to discuss. He seemed open to the request, but curious about more details and language considerations (what commitments are we making on how metadata is emitted?).

**Comment by teo-tsirpanis** [comment]
Would `CompilerLoweringPreserveAttribute` help here?
